cmd/cloud-controller-manager/main.go

```diff
@@ -41,16 +44,13 @@ import (
 	genericcontrollermanager "k8s.io/controller-manager/app"
 	"k8s.io/klog/v2"
 	nodeipamconfig "k8s.io/kubernetes/pkg/controller/nodeipam/config"
+	// For existing cloud providers, the option to import legacy providers is still available.
```

Comment:
So its not just in-tree providers. Out of tree providers can do the same things, however the import would obviously be to where ever they have their cloud provider implementation. For example Huawei has their implementation in https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/tree/master/pkg/cloudprovider/huaweicloud.